src/dbnode/persist/fs/retriever_test.go

```diff
@@ -303,13 +418,56 @@ func testBlockRetrieverHighConcurrentSeeks(t *testing.T, shouldCacheShardIndices
 			retrievedIDsMutex.Unlock()
 			require.True(t, ok, fmt.Sprintf("expected %s to be retrieved, but it was not", id))
 
-			expectedTags := ident.NewTags(testTagsFromTestID(id)...)
+			// Strip the volume tag because these reads were performed while concurrent block lease updates
+			// were happening so its not deterministic which volume tag they'll have at this point.
+			tags = stripVolumeTag(tags)
+			expectedTags := stripVolumeTag(testTagsFromIDAndVolume(id, 0))
 			require.True(
 				t,
 				tags.Equal(expectedTags),
 				fmt.Sprintf("expectedNumTags=%d, actualNumTags=%d", len(expectedTags.Values()), len(tags.Values())))
 		}
 	}
+
+	// Now that all the block lease updates have completed, all reads from this point should return tags with the
+	// highest volume number.
+	for _, shard := range shards {
+		for _, blockStart := range blockStarts {
+			for _, idString := range shardIDStrings[shard] {
+				var (
+					ctx = context.NewContext()
+					id  = ident.StringID(idString)
+				)
+
+				for {
+					// Run in a loop since the open seeker function is configured to randomly fail
+					// sometimes.
+					_, err := retriever.Stream(ctx, shard, id, blockStart, onRetrieve, nsCtx)
```

Comment:
super nit: Since you are now calling `BlockingClose()` below you should also call `ctx.Reset()` before using.